nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,249 @@ def save(self, *args, **kwargs):
     def delete(self, *args, **kwargs):
         # TODO: should we behave like CustomFieldChoice and raise a ProtectedError if this choice is in use anywhere?
         super().delete(*args, **kwargs)
+
+
+class ObjectMetadataManager(BaseManager.from_queryset(RestrictedQuerySet)):
+    use_in_migrations = True
+
+    def get_for_model(self, instance):
+        """Return all ObjectMetadatas assigned to the given instance."""
+        cache_key = f"{self.get_for_model.cache_key_prefix}.{instance}"
+        queryset = cache.get(cache_key)
+        if queryset is None:
+            assigned_object_type = ContentType.objects.get_for_model(instance)
+            assigned_object_id = instance.pk
+            queryset = self.get_queryset().filter(
+                assigned_object_type=assigned_object_type, assigned_object_id=assigned_object_id
+            )
+            cache.set(cache_key, queryset)
+        return queryset
+
+    get_for_model.cache_key_prefix = "nautobot.extras.objectmetadata.get_for_model"
+
+
+@extras_features(
+    "custom_validators",
+    "graphql",
+    "webhooks",
+)
+class ObjectMetadata(ChangeLoggedModel, BaseModel):
+    metadata_type = models.ForeignKey(
+        to=MetadataType,
+        on_delete=models.PROTECT,
+        related_name="object_metadatas",
+    )
+    contact = models.ForeignKey(
+        to=Contact,
+        on_delete=models.PROTECT,
+        related_name="object_metadatas",
+        blank=True,
+        null=True,
+    )
+    team = models.ForeignKey(
+        to=Team,
+        on_delete=models.PROTECT,
+        related_name="object_metadatas",
+        blank=True,
+        null=True,
+    )
+    scoped_fields = JSONArrayField(
+        base_field=models.CharField(
+            max_length=CHARFIELD_MAX_LENGTH,
+        ),
+        help_text="List of scoped fields, only direct fields on the model",
+    )
```

Comment:
Do we want/need to enforce that any specified scoped_fields actually exist on the given associated_object?

Need to think about how this will work for custom fields (and *maybe* relationships too?)

Reply:
https://github.com/nautobot/nautobot/issues/5861